Src/EB/AMReX_EB_geom.cpp

```diff
@@ -0,0 +1,324 @@
+            c_vec_tmp[d] = edge_p0[d] + lambda_tmp*edge_v[d];
+            rc_vec[d] = c_vec_tmp[d] - r_vec[d];
+        }
+
+        Real min_dist_tmp = dot_3d_real(rc_vec, rc_vec);
+
+        if (min_dist_tmp < min_dist) {
+            min_dist = min_dist_tmp;
+            for (int d=0; d<AMREX_SPACEDIM; ++d)
+                c_vec[d] = c_vec_tmp[d];
+        }
+    }
+
+    return c_vec;
+}
+
+
+
+void closest_dist (Real & min_dist, bool & proj_valid,
+                   const Vector<Real> & eb_data,
+                   const RealVect & dx_eb, const RealVect & pos) {
+
+    int l_eb = eb_data.size();
+
+    RealVect inv_dx;
+    for (int d=0; d<AMREX_SPACEDIM; ++d)
+        inv_dx[d] = 1. / dx_eb[d];
+
+    min_dist   = std::numeric_limits<Real>::max();
+    proj_valid = false;
+
+    Real min_dist2 = std::numeric_limits<Real>::max();
+    int i_nearest  = 0;
+
+
+    for (int i=0; i<l_eb; i+=2*AMREX_SPACEDIM) {
+
+        RealVect eb_cent, eb_norm;
+        for (int d=0; d<AMREX_SPACEDIM; ++d) {
+            eb_cent[d] = eb_data[i + d];
+            eb_norm[d] = eb_data[i + d + AMREX_SPACEDIM];
+        }
+
+        Real dist2 = dot_3d_real( pos - eb_cent, pos - eb_cent );
+
+        if ( dist2 < min_dist2 ) {
+            min_dist2 = dist2;
+            i_nearest = i;
+        }
+    }
+
+    RealVect eb_cent, eb_norm;
+    for (int d=0; d<AMREX_SPACEDIM; ++d) {
+        eb_cent[d] = eb_data[i_nearest + d];
+        eb_norm[d] = eb_data[i_nearest + d + AMREX_SPACEDIM];
+    }
+
+    Real dist_proj = dot_3d_real( pos - eb_cent, -eb_norm );
+
+    RealVect eb_min_pt;
+    IntVect  vi_cent, vi_pt;
+    for (int d=0; d<AMREX_SPACEDIM; ++d) {
+        eb_min_pt[d] = pos[d] + eb_norm[d] * dist_proj;
+
+        vi_cent[d] = static_cast<int>(amrex::Math::floor( eb_cent[d] * inv_dx[d]));
+        vi_pt[d]   = static_cast<int>(amrex::Math::floor( eb_min_pt[d] * inv_dx[d]));
+    }
+
+
+    bool min_pt_valid = false;
+    if ( all_eq( vi_pt, vi_cent ) ) {
+        min_pt_valid = true;
+    } else {
+        for (int k_shift = -1; k_shift <= 1; ++k_shift) {
+            for (int j_shift = -1; j_shift <= 1; ++j_shift) {
+                for (int i_shift = -1; i_shift <= 1; ++i_shift) {
+
+                    RealVect shift {AMREX_D_DECL(1.e-6*i_shift,
+                                                 1.e-6*j_shift,
+                                                 1.e-6*k_shift)};
+
+                    for (int d=0; d<AMREX_SPACEDIM; ++d)
+                        vi_pt[d] = static_cast<int>(amrex::Math::floor(
+                                (eb_min_pt[d] + shift[d]*dx_eb[d]) * inv_dx[d]
+                            ));
+
+                    if ( all_eq( vi_pt, vi_cent ) ) min_pt_valid = true;
+                }
+            }
+        }
+    }
+
+
+    if ( min_pt_valid ) {
+        min_dist   = dist_proj;
+        proj_valid = true;
+    } else {
+        RealVect c_vec = facets_nearest_pt(vi_pt, vi_cent, pos, eb_norm, eb_cent, dx_eb);
```

Comment:
Should `vi_pt` be reset to its original value before the i, j, k shift loops?

Reply:
That's a really good point! You would think so, but the original (fortran) code didn't have this either. I would call this a bug. Do you need me to fix it (i.e. check if it breaks anything when we reset `vi_pt`)?

Reply:
We do not need to do it now.  I actually have stolen your code and reimplemented it in #1425 to provide what mfix currently needs.  Since you have offered to move the LSCore code into your own code before, we plan to remove the LSCore code from the repo.  I am going to submit a PR probably this afternoon.  You will see what files and functions we plan to remove so that you can move them to your code.  Does that sound good?